alex/feat/cv bridge

ErLex20 opened the pull request:
@robmasocco take a look when you have time and tell me if I need to add more utilities.

ErLex20 committed:
chore: README updated

ErLex20 committed:
chore: targets created

ErLex20 committed:
feat: code base organization

ErLex20 committed:
feat: added frame_to_msg and msg_to_frame

ErLex20 committed:
feat: msg_to_frame fixed with ConstSharedPtr

ErLex20 committed:
feat: changed from class to just functions

ErLex20 committed:
feat: .hpp fixed

ErLex20 committed:
feat: added 16 bit types

ErLex20 committed:
feat: dua_msg_to_frame accepts cv::Mat & out_frame so overall only one .clone() is required

ErLex20 commented:
@robmasocco Optimization implemented.

robmasocco commented:
@ErLex20 dua-utils repo must be handled a little differently.
- They must be public since they are a part of the core of the DUA project.
- They should not be based on dua-template but only contain source code, properly organized in packages ready to be built.

robmasocco committed:
fix: final draft

@ErLex20 removed dua_* prefix in API names, it is already in the namespace name so it is redundant. The rest is good.
